ui/src/preview/preview-table.tsx

```diff
@@ -59,22 +63,22 @@ export enum PreviewSelectType {
 
 type PreviewTableProps = {
     previewChanges: PreviewCandidates;
-    onPreviewSelect: (selectedChanges: SelectedPublishChange) => void;
-    onRevert: (entry: PreviewTableEntry) => void;
     staged: boolean;
     changesBeingReverted?: ChangesBeingReverted;
     onShowOnMap: (bbox: BoundingBox) => void;
     changeTimes: ChangeTimes;
+    publicationAssetChangeAmounts: PublicationAssetChangeAmounts;
+    previewOperations: PreviewOperations;
```

Comment:
Passaillaan taas vaan objektin sisällä useampi funkkari (jotka voisivat siellä reduksissa elää). Ei tarvita siis kuuden eri funkkarireferenssin passausta tänne ja tämän lapselle.

Mielipiteitä rakenteesta? Periaatteessa voisi sisältää myös tuon onShowOnMapin, mutta pidin sen toistaiseksi erillisenä. Se olisi jo myös mahdollista noutaa delegaattina.

Reply:
Koitin lueskella vähän edestakaisin ja en ole ihan varma yksinkertaistaako tämä loppujen lopuksi vai niputtaako vain argumentteja olion kautta (mikä monimutkaistaa niiden lisäämistä/lukemista). Sanoisin kuitenkin että on tämä oikeaan suuntaan, koska tuo tuo niputus myös luo deklaratiivisen esityksen noille operaatioille ja se helpottaa hahmotusta kun niiden määrä lienee joka tapauksessa lisääntymään päin.

En ehkä myöskään heittäisi sinne mukaan showOnMap:ia koska se tuntuu jo ihan aika asialta.

Koska noiden käyttö meinaa venähtää pitkiksi riveiksi, tipauttaisin ehkä tuosta operaatioiden muuttujanimestä preview-sanan pois: se näkyy jo tyyppinimessä ja kyse on previewtaulukosta ja sen alakomponenteista, joten konteksti on minusta aika ilmeinen.